### ISSUE TITLE
[hackc] PHP classes named Container are broken

### ISSUE THREAD

**Issue by mgdigital** [facebook/hhvm#8204]
### HHVM Version
3.26.2

### Operating System and Version
Debian Wheezy

### Standalone code, or other way to reproduce the problem
If we have the PHP class:
```
<?php

namespace ContainerTest;

class Container {
    public function hello(): string {
        return 'hello';
    }
}
```
Then run the following code
```
<?hh

$c = new \ContainerTest\Container();
echo $c->hello();

```

### Expected result
"hello" should be output

### Actual result
Fatal error: Uncaught Error: unknown class Container

The error does not happen when `hhvm.hack_compiler_default=false` is added in php.ini

**Comment by fredemmott** [comment]
This is not reproducible on 3.26.0, 3.26.1, or 3.26.2.

I am using:

```
fredemmott-pro:tmp fredemmott$ cat a.php
<?hh

require_once('test.php');

$c = new \ContainerTest\Container();
echo $c->hello();
fredemmott-pro:tmp fredemmott$ cat test.php
<?php

namespace ContainerTest;

class Container {
    public function hello(): string {
        return 'hello';
    }
}
```

**Comment by fredemmott** [comment]
You probably want to follow-up with whoever is providing your packages; 3.25 and above are not supported on Wheezy, and we don't build packages for it any more.

**Comment by mgdigital** [comment]
Okay I'm running the official hhvm/hhvm:3.26.2 Docker image now and the behaviour seems slightly different but I can still reproduce this as follows:

```
$ cat Container.php
<?php

namespace ContainerTest;

class Container {
    public function hello(): string
    {
        return 'hello';
    }
}

$ cat ContainerExt.php
<?php

namespace ContainerTest;

class ContainerExt extends Container {
}

$ cat ContainerTest.hh
<?hh

require_once __DIR__ . '/Container.php';

var_dump(class_exists(\ContainerTest\Container::class));

require_once __DIR__ . '/ContainerExt.php';

$container = new \ContainerTest\ContainerExt();
echo $container->hello();

$ hhvm ContainerTest.hh
bool(true)

Fatal error: Uncaught Error: unknown class Container in /app/ContainerExt.php:5
Stack trace:
#0 /app/ContainerTest.hh(4): include()
#1 {main}
```

**Comment by fredemmott** [comment]
Thanks, can repro now. To make the example a bit smaller:


```
$ cat ContainerExt.php
<?php

namespace ContainerTest;

class ContainerExt extends Container {
}

fredemmott-pro:hhtest fredemmott$ cat ContainerExt.php
<?php

namespace ContainerTest;

class ContainerExt extends Container {
}
```

```
$ hhvm -v Eval.HackCompilerDefault=false -v Eval.DumpHhas=1 ContainerExt.php > hphpc.hhas
$ hhvm -v Eval.HackCompilerDefault=true -v Eval.DumpHhas=1 ContainerExt.php > hackc.hhas
$ diff -u hackc.hhas hphpc.hhas
--- hackc.hhas	2018-05-18 08:12:50.000000000 -0700
+++ hphpc.hhas	2018-05-18 08:12:43.000000000 -0700
@@ -7,13 +7,13 @@
 .hh_file 0;

 .main (1,1) {
-  .srcloc 5:7,5:18;
+  .srcloc 5:1,5:1;
   DefCls 0
   Int 1
   RetC
 }

-.class ContainerTest\ContainerExt extends Container {
+.class ContainerTest\ContainerExt extends ContainerTest\Container {
 }

 # /private/tmp/hhtest/ContainerExt.php ends here
```

**Comment by fredemmott** [comment]
If I change the class to `extends ContainerFoo`, hackc and hphpc both generate `.class ContainerTest\ContainerExt extends ContainerTest\ContainerFoo {`

**Comment by fredemmott** [comment]
Also, the repro does not require a mix of `<?hh` and `<?php` - it also reproduces with just PHP code.

**Comment by fredemmott** [comment]
Should be fixed in 3.26.3 , and hopefully today's nightlies
